store/local/job_spec_adapter.go

```diff
@@ -568,8 +530,19 @@ func (JobSpecAdapter) FromSpec(spec models.JobSpec) (Job, error) {
 		})
 	}
 
+	var truncateTo, offset, size string
+	if spec.Task.Window != nil {
+		truncateTo = spec.Task.Window.GetTruncateTo()
+		offset = spec.Task.Window.GetOffset()
+		size = spec.Task.Window.GetSize()
+	}
+
+	version := spec.Version
+	if version == 0 {
```

Comment:
this should be handled within the constructor

Reply:
this is in case the job spec version is not specified. it's like a default value. the constructor's responsibility is construct a window based on valid inputs, or return error otherwise. if there's any invalid input such as version, I believe the constructor shouldn't assume the version of a spec. if there's a strong reason to do otherwise, then we need to at least change the function name to be not `NewWindow`